src/main/java/com/enderio/base/common/menu/EntityFilterSlot.java

```diff
@@ -0,0 +1,74 @@
+package com.enderio.base.common.menu;
+
+import com.enderio.api.capability.StoredEntityData;
+import com.enderio.base.common.init.EIOCapabilities;
+import net.minecraft.client.Minecraft;
+import net.minecraft.world.Container;
+import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.SpawnEggItem;
+
+import java.util.function.Consumer;
+
+public class EntityFilterSlot extends Slot {
+
+    private static Container emptyInventory = new SimpleContainer(0);
+    private final Consumer<StoredEntityData> consumer;
+
+    public EntityFilterSlot(Consumer<StoredEntityData> consumer, int pSlot, int pX, int pY) {
+        super(emptyInventory, pSlot, pX, pY);
+        this.consumer = consumer;
+    }
+
+    @Override
+    public ItemStack getItem() {
+        return ItemStack.EMPTY;
+    }
+
+    @Override
+    public void set(ItemStack pStack) {
+        setChanged();
+    }
+
+    @Override
+    public void setChanged() {
+
+    }
+
+    @Override
+    public ItemStack remove(int pAmount) {
+        set(ItemStack.EMPTY);
+        return ItemStack.EMPTY;
+    }
+
+    @Override
+    public int getMaxStackSize() {
+        return getItem().getMaxStackSize();
+    }
+
+    @Override
+    public ItemStack safeInsert(ItemStack stack, int amount) {
+        // If this stack is valid, set the inventory slot value.
+        if (stack.isEmpty() || !mayPlace(stack)) {
+            return stack;
+        }
+
+        if (stack.getCapability(EIOCapabilities.ENTITY_STORAGE).isPresent()) {
```

Comment:
Well yeah that definitely looks unoptimized right away. Was I drunk or smth hahah. Fixing it in a minute also!

Reply:
Okay I can now remember why I did that. This is due to the fact that the get() method is private within LazyOptional on 1.20.1 forge so I have to use orElseGet() or orElse() method which are both doing checkups once again, in the end making the optimization level exactly the same as this (if not slightly worse). However, if you'd like, I can make a version with orElse() for readability reasons with a warning added due to usage of null on a @NotNull annotated object.

Reply:
Yes I have thought about this for half an hour straight and every approach would be the same (if the others are not slightly worse) because of the LazyOptional's implementation on 1.20.1. Both orElse(), orElseGet() and resolve() are alternatives to get(). However, all of these make a couple of null and validity checks before returning the value. Therefore, making them almost on par with my isPresent()+ifPresent() approach. However, if you have any other idea, I am all ears.

Reply:
Well I have decided to make an accurate calculation by time complexity analysis and here are the results:

Definitions:
n -> Time complexity of a single comparison
m -> Time complexity of any minor action on objects (requireNonNull(), of() for example) (the easy part)
k -> Time complexity of getValue() inside an optional (the worst part)

Approaches:
1. isPresent() + ifPresent() (current approach)
2. resolve() + isEmpty() + get()
3. orElse(null) + null check

Time complexities:
1. 5n+m+k = O(n+m+k) (if we want to look at this from a big-o notation like perspective)
2. 6n+2m+k = O(n+m+k))
3. 2n+k = O(n+k)

Conclusion:
Second option and first option (current) is very close to each other with the current one being somewhat better. However, the third approach seems to be a much better option with a minor drawback. The third option will give out 2 warnings because of  the usage of null in a Non-Null context, however it won't be a problem otherwise.

Side note: I know that this was an overkill analysis for, at most, a very slight performance optimization but I had nothing else to do hahah

Tl;dr Found a better solution then isPresent+ifPresent but it will add 2 warnings, would that be a problem for you guys?

Reply:
Tl;dr Found a better solution then isPresent+ifPresent but it will add 2 warnings, would that be a problem for you guys?

Well I have decided to make an accurate calculation by time analysis and here are the results:

Definitions:
n -> Time taken for a single comparison
m -> Time taken for any minor action on objects (the easy part)
k -> Time taken for getValue() inside an optional (the worst part)

Approaches:
1. isPresent() + ifPresent() (current approach)
2. resolve() + isEmpty() + get()
3. orElse(null) + null check

Times taken:
1. 5n+m+k
2. 6n+2m+k
3. 2n+k

Conclusion:
Second option and current approach is very close to each other with the current one being somewhat better. However, the third approach seems to be a much better option with a minor drawback. The third option will give out 2 warnings because of  the usage of null in a Non-Null context, however it won't be a problem otherwise.

Side note: I know that this was an overkill analysis for, at most, a very slight performance optimization but I had nothing else to do hahah